llvm/lib/Transforms/Vectorize/LoopVectorize.cpp

```diff
@@ -4075,7 +4075,12 @@ LoopVectorizationCostModel::computeMaxVF(ElementCount UserVF, unsigned UserIC) {
     unsigned MaxVFtimesIC =
         UserIC ? *MaxPowerOf2RuntimeVF * UserIC : *MaxPowerOf2RuntimeVF;
     ScalarEvolution *SE = PSE.getSE();
-    const SCEV *BackedgeTakenCount = PSE.getBackedgeTakenCount();
+    // Currently only loops with countable exits are vectorized so it's safe to
+    // use getSymbolicMaxBackedgeTakenCount as it should give the same result
+    // as getBackedgeTakenCount.
```

Comment:
To me the comment suggests when uncountable exists are support calling `getSymbolicMaxBackedgeTakenCount` becomes unsafe?, which I think is the opposite of what you mean.

Perhaps something akin to:

Currently only loops with countable exits are vectorized, but calling `getSymbolicMaxBackedgeTakenCount` allows enablement work for loops with uncountable exits whilst also ensuring the symbolic maximum and known back-edge taken count remain identical for loops with countable exits.

Reply:
Seems like a good suggestion! I've rebased the patch to see if the libc++ tests start passing now, and updated the comment. Two for the price of one. :)